src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -490,6 +491,16 @@ def join(
             source_plan,
         )
 
+    def _table_exists(self, table_name):
+        try:
+            # DESC is used here because SHOW does not work for qualified table name
+            self.session.sql(f"DESC TABLE {table_name}").collect()
+            return True
+        except SnowparkSQLException:
+            # We could verify if this is a compile error and throw otherwise, but directly return False gives no
+            # behavior change as the old behavior assumes table does not exist
+            return False
```

Comment:
To be honest I'm not sure if using `DESC` is a good idea or not as it requires a try-catch. Using `SHOW` is not straightforward as we need to extract database and schema from the table_name. It could easily get messy when there are dots in the double quoted identifier